### PR TITLE
xmeta design notes draft

### PR DESCRIPTION
**Pull request by timr1126**
This is an initial draft of the design notes. Will need to elaborate, but the basics of xmeta are there.

### CODE REVIEW

File: design_notes/XDN13 - xmeta Architecture and Purpose.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+## Abstract
+
+This document will outline the purpose of xmeta and the design and architecture decisions made for the tool.
+
+## Purpose
+
+The xmeta tool will be a cross platform version of MIDLRT. As a brief sidenote, MIDLRT was a fork of MIDL with a very verbose syntax. A new syntax called MIDL3 was created, and it was added to MIDLRT rather than having its own tool built from scratch. Unfortunately, there is now a lot of hardcoded WinRT knowledge in MIDLRT that would be difficult to adapt for xlang.
+
+Functionally, xmeta will take in a MIDL3 [IDL](https://docs.microsoft.com/en-us/windows/desktop/com/idl-files) file and previously compiled `.winmd` files, and will compile them into new `.winmd` metadata files.
```
Comment (BenJKuhn) [inline design_notes/XDN13 - xmeta Architecture and Purpose.md]: General question: are we keeping the same extension for xlang metadata? That seems potentially troublesome.

Comment (timr1126) [reply design_notes/XDN13 - xmeta Architecture and Purpose.md]: We will be renaming the extension. Probably to `.xmeta`.

Comment (jkoritzinsky) [reply design_notes/XDN13 - xmeta Architecture and Purpose.md]: Also, will there be a bit set in the metadata to differentiate between a WinMD and an XMeta file like how there's currently the `tdWindowsRuntime` bit to differentiate a WinRT class and a .NET class?

Comment (BenJKuhn) [reply design_notes/XDN13 - xmeta Architecture and Purpose.md]: We've discussed this bit from time to time, but haven't made a decision yet. One key data point is that although we're leveraging the .NET metadata format to describe types, we expect these will likely NOT be read by the CLR directly the way winrt types are. The xlang C# projection will likely be independent of the .NET core code, which makes this questions a bit less critical.

Comment (jkoritzinsky) [reply design_notes/XDN13 - xmeta Architecture and Purpose.md]: Thanks for the response! That's good to know!

**Pull request merged**